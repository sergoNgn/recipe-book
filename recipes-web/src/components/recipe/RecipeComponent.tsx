import { Card, CardContent, CardMedia, Typography } from "@mui/material";
import { FC } from "react";
import { Recipe } from "../../types";

interface RecipeComponentProps {
  recipe: Recipe;
  onClick: (recipeId: number | undefined) => void;
}

const RecipeComponent: FC<RecipeComponentProps> = ({ onClick, recipe }) => {
  return (
    <Card
      className="cursor-pointer hover:shadow-lg h-54 sm:w-full md:w-2/5 xl:w-64"
      onClick={() => onClick(recipe.id)}
    >
      <CardMedia
        component="img"
        sx={{ height: 80 }}
        image="/images/recipe.jpg"
        alt="recipe image"
      />
      <CardContent className="select-none">
        <Typography
          gutterBottom
          variant="subtitle1"
          className="truncate"
          component="div"
          title={recipe.name}
        >
          {recipe.name}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          {recipe.shortDescription}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default RecipeComponent;
