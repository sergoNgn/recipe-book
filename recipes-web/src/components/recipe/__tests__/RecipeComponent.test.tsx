import { render } from "@testing-library/react";
import RecipeComponent from "../RecipeComponent";
import { recipe } from "./__mocks__";

describe("Recipe Component", () => {
  it("renders recipe component", async () => {
    const recipeComponent = render(
      <RecipeComponent recipe={recipe} onClick={() => {}} />
    );

    expect(recipeComponent.baseElement).toMatchSnapshot();
  });
});
