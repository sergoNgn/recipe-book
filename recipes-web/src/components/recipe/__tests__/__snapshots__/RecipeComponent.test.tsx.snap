// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Component renders recipe component 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root cursor-pointer hover:shadow-lg h-54 sm:w-full md:w-2/5 xl:w-64 css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <img
        alt="recipe image"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-rgbcif-MuiCardMedia-root"
        src="/images/recipe.jpg"
      />
      <div
        class="MuiCardContent-root select-none css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom truncate css-1wniyei-MuiTypography-root"
          title="recipe #1"
        >
          recipe #1
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          short description
        </p>
      </div>
    </div>
  </div>
</body>
`;
